### PR TITLE
feat(HMS-1429): Add get gcp reservation by id 

### PR DESCRIPTION
**Pull request by adiabramovitch**
This PR is Adding get GCP reservation by id as we have for Azure and for AWS.

### CODE REVIEW

File: internal/payloads/reservation_payload.go
Comment (lzap) [inline internal/payloads/reservation_payload.go:139]: I will make sense to create struct here instead of string. It is very likely that we will want to also fetch DNS and IP of the launched instances alongside id and refactoring later is painful, as you can see here: https://github.com/RHEnVision/provisioning-backend/pull/448/files#diff-93db714a65f393cc33692e57cab38ad7ae1b8b12b3a0cefb1769b435b747a0d1

For now, I suggest to create a struct type with just a string as the single field so we can add IP/DNS later.

**Review by lzap — CHANGES_REQUESTED** [review]
Perfect, tho, to prevent a problem with future string to struct change I suggest to create a struct already now so we can add the remaining fields later in the future.

File: internal/payloads/reservation_payload.go
Comment (adiabramovitch) [reply internal/payloads/reservation_payload.go:139]: I see that Amir has added `InstanceResponse` type for AWS. 
I guess I should wait until his PR will be merged?

File: internal/payloads/reservation_payload.go
Comment (lzap) [inline internal/payloads/reservation_payload.go:48]: This will cause conflict with @amirfefer 's PR but this will likely go in earlier because there is no IQE test breakage here. Sorry :-)

File: api/openapi.gen.yaml
Comment (lzap) [inline api/openapi.gen.yaml:642]: This field is exposed for no reason tho, I mean this is reservation response, there is no need to copy reservation_id N times, it has zero information since the user performed GET call with reservation ID already.

Now since any refactoring would cause pains, what I suggest is to merge this as is and ask @amirfefer to hide this field (by simply copying only relevant fields into the payload type from model type).

**Review by lzap — APPROVED** [review]
TY I will explain to Amir what to do.

**Review by lzap — CHANGES_REQUESTED** [review]
Ok since instance detail is higher priority than GCP, I take back what I said - I will ask Amir to do this small change first, merge his PR since IQE tests are almost ready and then I will ask you to rebase on top of that. Will be just a tiny conflict for the `InstanceResponse` type.

**Comment by lzap** [comment]
@adiabramovitch we have merged @amirfefer 's PR. In case you are wondering what was the change, it was simple: https://github.com/RHEnVision/provisioning-backend/compare/d53b532f924f75bd0198456b7d36c81bf7857843..27bcce0c7cffb1dcdd20e3df3b23e9eb39b46e5d

Please rebase this PR is ready IMHO. Thank you.

**Review by ezr-ondrej — APPROVED** [review]
Yup, looks good :+1:
I guess this should wait for @akhil-jha to prepare IQE tests (btw we can help if that would help, given we already have credentials).

Thanks for fixing the Azure wording :orange_heart:

File: internal/payloads/reservation_payload.go
Comment (ezr-ondrej) [inline internal/payloads/reservation_payload.go:46]: We should probably do the same what we did in Amir's PR and for the Instances response (copy the instance details), but I guess that will be done while rebasing.

Comment (adiabramovitch) [reply internal/payloads/reservation_payload.go]: rebased

**Comment by lzap** [comment]
Re: Unleash. I think it is fine not to have it any only hide the UI part, @ezr-ondrej

**Comment by lzap** [comment]
A conflict again. Can you rebase?

Shall we just merge this since GCP is not yet exposed to the users without automation? What harm can it done, GCP code is already in. @ezr-ondrej @akhil-jha

**Comment by lzap** [comment]
Errors with:

AttributeError: 'ProvisionedHost' object has no attribute 'client'

**Comment by lzap** [comment]
Fails for my PR as well, pinged QA folks what is going on :-)

**Review by lzap — APPROVED** [review]
All good.

I suggest to merge this PR which introduces OpenAPI for GCP even if we are not yet ready to write a IQE test. If we end up need to push to prod without GCP, we can add an unleash flag so the reservation returns an error. @akhil-jha

File: cmd/spec/main.go
Comment (lzap) [reply cmd/spec/main.go:27]: Correction: I suggest to split OpenAPI from this PR and merge without it.

**Pull request merged**
